pkg/routes/linux_routes.go

```diff
@@ -30,3 +32,115 @@ func DeleteByDestination(destinationSubnet *net.IPNet) error {
 	}
 	return nil
 }
+
+type LinuxRouter struct {
+	NodeIPA       pkg.NodeIPAware
+	Tunneler      pkg.Tunneler
+	RouteSyncer   pkg.RouteSyncer
```

Comment:
I don't particularly like the circular dependency here. `LinuxRouter` requires `pkg.RouteSyncer` so that it can update the state host routes. `pkg.RouteSyncer` in turn requires `LinuxRouter` so that it can reuse the `InjectRoute()` logic when new routes are found during `checkCacheAgainstBGP()`.

It feels like there should be a better / cleaner way to do this.